src/controller/CommissioningDelegate.h

```diff
@@ -376,12 +376,9 @@ class CommissioningParameters
         return *this;
     }
 
-    // If a ThreadOperationalDataset is provided, then the ThreadNetworkScan will not be attempted
     CommissioningParameters & SetThreadOperationalDataset(ByteSpan threadOperationalDataset)
     {
-
         mThreadOperationalDataset.SetValue(threadOperationalDataset);
-        mAttemptThreadNetworkScan = MakeOptional(static_cast<bool>(false));
```

Comment:
This seems wrong; if we have thread credentials we shouldn't _also_ do a network scan.

If the issue is that state that is persisted across commissionings is not reset properly when a commissioning ends, then _that_ is what needs to get fixed.

Reply:
@bzbarsky-apple 
no. I don't think so.
Since this is something that is explicitly set by the user who actually uses this parameter, I think it would cause more confusion if it was automatically disabled when the dataset is set.

I would agree with you if it were to be disabled due to a change in behavior, but this is a situation where the direct user has explicitly enabled it. For Wi-Fi, no such thing is set up.

Reply:
i concur with @bzbarsky-apple ,  if we have thread credentials we shouldn't also do a network scan.

Reply:
@yunhanw-google 
Enabling network scan is intentional by those using the API. In addition, network credentials can be set not only at the beginning, but also after a network scan, and there is a problem in that network scans cannot be performed again after this. There is no other way, so how can we solve this issue?

I think saving it somewhere else and then recovering it is a more complicated approach.

With scan enabled, I think it's natural to add credentials for thread connection.

Reply:
If we think the users should not enable network scan, I would suggest making use of the Optional:

- Not set: Do a scan if no credentials provided
- True: Always scan
- False: Always not to scan.

I think we should also clearify what this function really does, according to its name `CommissioningParameters`, it seems to be a configuration, but if we have a fuse `mAttemptThreadNetworkScan = MakeOptional(static_cast<bool>(false));` then it more like a state?

Reply:
@bzbarsky-apple 
https://github.com/project-chip/connectedhomeip/commit/5706a210f9220e8761bd5f944678efd0e4fe2c89

Here, is there a reason why the three parameter settings below were deleted?
AttemptThreadNetworkScan
AttemptWiFiNetworkScan
SkipCommissioningComplete

Even if you save the Commissioning Parameter Set in the Upper Layer, it is difficult to process it separately because you deleted some parts.

Reply:
> For Wi-Fi, no such thing is set up.

How does the Wi-Fi behavior differ?  I would expect the same behavior there: no scan if we have credentials....  But maybe I'm missing something about how this is supposed to work?

What is the intended behavior of "scan + credentials" and what are the credentials used for in that case?

> Here, is there a reason why the three parameter settings below were deleted?

I don't follow the question.  What was deleted?  If you're asking why we are no longer getting/setting those explicitly, it's because the `mParams = params;` that changeset added handles all that for simple boolean bits.

Reply:
@bzbarsky-apple 
The problem now is as follows.
- If you add Thread Credentials after ScanNetwork, ScanNetworks will no longer work.
   -> Settings for ScanNetwork are not setting each commissioning, but only once when the DeviceCommissioner object is created.

The reason why I asked why the three parameters above were deleted is because I think there is a need to restore that part in order to manage the Commissioning Parameter separately (params = mParam).

  As you said, Wi-Fi and Thread must be managed the same way. (A decision must be made whether to delete both or add both.)

There are two solutions I thought of.
1. Similar to the PR I currently uploaded, when setting Credentials, disabling ScanNetwork was deleted. (Of course, I know your opinion is No.)

2. Manage the Commissioning Parameters separately and set them in the Commissioning Parameters in AutoCommissioner when commissioning.
  -> In this case, as mentioned above, the code related to the three parameter settings must be restored.

Reply:
> Settings for ScanNetwork are not setting each commissioning, but only once when the DeviceCommissioner object is created.

If that is the case, that would be the bug that needs fixing, no?

>  In this case, as mentioned above, the code related to the three parameter settings must be restored.

Which exact code?  Again, those settings are copied from the incoming parameters by the assignment to `mParams`....

Reply:
@bzbarsky-apple 
I restored the code and separated the part that sets the Android Commissioning Parameter. Is this the right way to make changes?